30 add structure current loc

jasonrogena opened the pull request:
Closes #30

jasonrogena committed:
JAR: Pan to User's Location When Fix First Gotten

Pan to a user's location when a fix is first gotten after the GPS
countdown dialog has been closed. Part of fix for issue #30

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
JR: LocationListeners in MapView

Add support for adding extra LocationListeners in the MapView's
GpsLocationProvider. Part of fix for issue #30

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
JR: Implement Add Node Where You Stand

Make it such that users are only allowed to add nodes at their current
GPS location. Feature can be turned off using the
proximity_settings.xml. Fix for issue #30

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
JR: Tests for Issue #30 Fix

Add tests for the code that fixes issue #30

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

ukanga commented on MapboxAndroidSDK/src/main/java/com/mapbox/mapboxsdk/overlay/GpsLocationProvider.java:
```diff
@@ -58,19 +88,30 @@ public void setLocationUpdateMinDistance(final float meters) {
-                    mLocation = mLocationManager.getLastKnownLocation(provider);
-                    if (mLocation != null) {
-                        mMyLocationConsumer.onLocationChanged(mLocation, this);
+        if(testMode == true) {
```
This is one area that the code could be mocked, the else section is entirely skipped and it seems that would be the place that needed to be tested.

denniswambua commented on app/src/main/java/org/redcross/openmapkit/MapActivity.java:
```diff
@@ -1014,6 +1017,9 @@ private boolean isGPSEnabled() {
     }
 
     private void showProgressDialog() {
+        if(isUserLocationEnabled() == false) {
```
better use `if(!isUserLocationEnabled())`

jasonrogena replied:
Is there a specific reason why I should use '!' instead of false? I find using false better, especially when it comes to skimming through the code. Very hard to miss a false

denniswambua commented on app/src/main/java/org/redcross/openmapkit/MapActivity.java:
```diff
@@ -622,7 +598,22 @@ private void toggleNodeMode() {
+            if(LocationXMLParser.getProximityCheck()) {
+                //check user's last location is accurate enough
+                if(lastLocation != null && lastLocation.getAccuracy() <= LocationXMLParser.getGpsProximityAccuracy()) {
+                    if(isUserLocationEnabled() == false) {
```
same here `if(!isUserLocationEnabled())`

jasonrogena committed:
JR: Remove 'testMode' Boolean Var in MapActivity

Remove code dependency on the boolean variable testMode/isTesting in the
MapActivity class and other dependent classes. Part of fix for issue #30

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

denniswambua commented on MapboxAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/MapView.java:
```diff
@@ -1687,6 +1705,10 @@ private boolean handleTwoFingersTap(MotionEvent event) {
     @Override
     public boolean onTouchEvent(MotionEvent event) {
 //        Log.i(TAG, "onTouchEvent with event = " + event);
```
Commented out code. Just remove it

jasonrogena replied:
OK, cool. Heads up though, the commented out code is from [upstream](github.com/americanredcross/openmapkitandroid)

denniswambua commented on app/src/main/java/org/redcross/openmapkit/MapActivity.java:
```diff
@@ -650,22 +621,24 @@ public void onClick(View v) {
+        if(LocationXMLParser.getProximityCheck() == false) {
+            final ImageButton moveNodeModeBtn = (ImageButton)findViewById(R.id.moveNodeModeBtn);
+            final ImageButton moveNodeMarkerBtn = (ImageButton)findViewById(R.id.moveNodeMarkerBtn);
+            final Button moveNodeBtn = (Button)findViewById(R.id.moveNodeBtn);
```
Is this a boolean or a Button object?

jasonrogena replied:
@denniswambua the method returns a boolean

denniswambua commented:
code review :-1:

jasonrogena committed:
JR: Verbose Conditional Statements

Rewrite all new verbose conditional statements. Part of fix for issue

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

jasonrogena committed:
JR: Unused Logging Lines

Remove all unused logging lines. Part of fix for issue #30

Signed-off-by: Jason Rogena <jasonrogena@gmail.com>

denniswambua commented:
:+1:  lgtm
